sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -273,32 +278,71 @@ reported_pings AS (
     client_id
 ),
 _current AS (
+  -- Get first value when the same ping type returns more than one record with the exact same TIMESTAMP
```

Comment:
At this point the value is expected to be unique, however is safer to consider first in case the same ping type was sent more than once. Thoughts?

Reply:
I'm of two minds here. I prefer our ETLs to be completely idempotent. But this adds a good bit of complexity for a scenario that is quite possible to never occur. If we leave this part out, what happens when we receive two first pings of the same type from the same timestamp for a client?

Reply:
In that case the subquery of that ping type returns one record with the values depending on the order in which data is returned by that subquery. So two records for the same (client_id1, sample_id1, timestamp1) and different country would result in (client_id1, sample_id1, timestamp1, first_country_depending_on_the_order).

Reply:
I think that is acceptable in this case, given how rarely this should occur. Let's verify this with the team during the infra meeting tomorrow (thesis being: we should _almost_ never receive the same ping with the same timestamp from a client, though I suppose we can check this with a query).